core/Azure.Mcp.Core/src/Services/Http/HttpClientFactoryConfigurator.cs

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
+using Azure.Mcp.Core.Areas.Server.Options;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Http;
+using Microsoft.Extensions.Options;
+
+namespace Azure.Mcp.Core.Services.Http;
+
+public static class HttpClientFactoryConfigurator
+{
+    private static readonly string s_version;
+    private static readonly string s_framework;
+    private static readonly string s_platform;
+
+    static HttpClientFactoryConfigurator()
+    {
+        var assembly = typeof(HttpClientService).Assembly;
+        s_version = assembly.GetCustomAttribute<AssemblyFileVersionAttribute>()?.Version ?? "unknown";
+        s_framework = assembly.GetCustomAttribute<TargetFrameworkAttribute>()?.FrameworkName ?? "unknown";
+        s_platform = RuntimeInformation.OSDescription;
+    }
+
+    public static IServiceCollection ConfigureDefaultHttpClient(this IServiceCollection services)
+    {
+        ArgumentNullException.ThrowIfNull(services);
+
+        services.ConfigureHttpClientDefaults(static builder => ConfigureHttpClientBuilder(builder));
+
+        return services;
+    }
+
+    private static void ConfigureHttpClientBuilder(IHttpClientBuilder builder)
+    {
+        builder.ConfigureHttpClient((serviceProvider, client) =>
+        {
+            var httpClientOptions = serviceProvider.GetService<IOptions<HttpClientOptions>>()?.Value ?? new HttpClientOptions();
```

Comment:
Any reason why we're using `GetService` rather than `GetRequiredService`? Do we want to allow for a dependency to not exist? If so, please add a comment describing the atypical, but valid, leniency.

Reply:
Since DI can't be checked at compile time, I tend to prefer inflexibility. Silently falling back on uninitialized values leads to unknown bugs.